src/map/Map.js

```diff
@@ -252,6 +252,13 @@ export var Map = Evented.extend({
 	_getBoundsCenterZoom: function (bounds, options) {
 
 		options = options || {};
+
+		if ((options.padding && options.padding.length !== 2) ||

```

Comment:
Invalid value like '50' will pass this test too.

Reply:
This is just to catch cases where a single numeric value is passed in, as it was to deal with the original #7372 issue. Yes, a two-character string will be classed as valid, but I think that's beyond the scope of this simple fix.